### DISCUSSION TITLE
Alpine.js x-on:keyup.enter event not triggering with Windows Narrator ON

### DISCUSSION THREAD

**Discussion by HarieswaranD** [alpinejs/alpine#3991] [7. Bugs (archived)] [answered]
II have created a menu bar for the site [service.infiniti.ca](https://service.infiniti.ca/) using Alpine.js. The menu is designed to show and hide on the press of the enter key by using the x-on:keyup.enter="open = ! open" event. However, when the Windows Narrator is ON, the x-on:keyup event is not being triggered at all.

Steps to reproduce:

Implement a menu bar using Alpine.js with the x-on:keyup.enter="open = ! open" event for show/hide functionality.
Turn ON the Windows Narrator.
Press the enter key to trigger the menu show/hide functionality.
Expected behavior:
The menu should show/hide upon pressing the enter key, regardless of whether the Windows Narrator is ON or OFF.

Actual behavior:
The menu does not show/hide when the Windows Narrator is ON. The x-on:keyup event does not seem to be triggered.

Please suggest a solution or workaround to ensure that the menu show/hide functionality works as expected with the Windows Narrator ON.
![image](https://github.com/alpinejs/alpine/assets/81755830/7508f7bd-3313-4ddb-ab14-d0698b281b1b)

**Comment by ekwoka** [comment]
use `@keydown` instead.

That should do it.

**Reply by HarieswaranD** [reply]
No. its not working

**Reply by ekwoka** [reply]
Okay, well windows narrator uses spacebar and enter as shortcuts in scan mode.

So you need to provide a way that doesn't not use those. @click is a possibility, since that is what the action is that it will most likely do, or use @focus events, etc.

**Reply by HarieswaranD** [reply]
Thanks for your suggestion. I got it working by implementing a workaround. But its it an issue with Alpine Js?

**Reply by ekwoka** [reply, marked as answer]
No, it's what the tool does. It does not provide the key events for those two keys.

**Comment by HarieswaranD** [comment]
I have got it working now. If you look at the image below, there are two blocks. The first block is not working whereas 2nd one is working.  But still, I wonder why the first block of code is not working with Windows Narrator ON 

![image](https://github.com/alpinejs/alpine/assets/81755830/c4c2d5f1-cf58-4e23-9067-fdebafc7b503)

**Reply by ekwoka** [reply]
Because Narrator Scan uses the Enter and Space keystrokes to do primary action (click).

**Comment by ekwoka** [comment]
Also, btw, that site uses Alpine v2. This should REALLY be updated to v3 at least, ideally the latest...
